File: Wrapping/WrapITK/Java/Tests/BinaryThresholdImageFilter.java
/**
 *  Example on the use of the BinaryThresholdImageFilter
 *
 */

import InsightToolkit.*;

public class BinaryThresholdImageFilter
{
  public static void main( String argv[] )
  {
    System.out.println("BinaryThresholdImageFilter Example");

    itkImageFileReaderIUS2_Pointer reader = itkImageFileReaderIUS2.itkImageFileReaderIUS2_New();
    itkImageFileWriterIUS2_Pointer writer = itkImageFileWriterIUS2.itkImageFileWriterIUS2_New();

    itkBinaryThresholdImageFilterIUS2IUS2_Pointer filter = itkBinaryThresholdImageFilterIUS2IUS2.itkBinaryThresholdImageFilterIUS2IUS2_New();

    filter.SetInput( reader.GetOutput() );
    writer.SetInput( filter.GetOutput() );

    reader.SetFileName( argv[0] );
    writer.SetFileName( argv[1] );

    filter.SetLowerThreshold(  Integer.parseInt( argv[2] )  );
    filter.SetUpperThreshold(  Integer.parseInt( argv[3] )  );

    filter.SetOutsideValue( Integer.parseInt( argv[4] )  );
    filter.SetInsideValue(  Integer.parseInt( argv[5] )  );

    writer.Update();
  }

}

File: Wrapping/WrapITK/Python/Tests/BinaryThresholdImageFilter.py
#
#  Example on the use of the BinaryThresholdImageFilter
#

import itk
from sys import argv
itk.auto_progress(2)

dim = 2
IType = itk.Image[itk.US, dim]

reader = itk.ImageFileReader[IType].New( FileName=argv[1] )
filter  = itk.BinaryThresholdImageFilter[IType, IType].New( reader,
                LowerThreshold=eval( argv[3] ),
                UpperThreshold=eval( argv[4] ),
                OutsideValue=eval( argv[5] ),
                InsideValue=eval( argv[6] ))
writer = itk.ImageFileWriter[IType].New( filter, FileName=argv[2] )

writer.Update()

